babel/src/main/java/org/apache/calcite/sql/babel/SqlIndex.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.babel;
+
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import java.util.List;
+
+/**
+ * A <code>SqlIndex</code> is a class that can be used
+ * to create an index used by the SQL CREATE TABLE
+ * function.
+ */
+public abstract class SqlIndex {
+  public final SqlParserPos pos;
+  public final List<SqlNode> columns;
+  public final SqlIdentifier name;
+  public final boolean isUnique;
+
+  public SqlIndex(SqlParserPos pos, List<SqlNode> columns, SqlIdentifier name, boolean isUnique) {
+    this.pos = pos;
+    this.columns = columns;
+    this.name = name;
+    this.isUnique = isUnique;
+  }
+
+  public abstract void unparse(SqlWriter writer, int leftPrec, int rightPrec);
+
+  public SqlParserPos getParserPos() {
```

Comment:
Is this necessary if pos is already public?

Reply:
Maybe the above variables could be made private and have getters to access them in the child class

Reply:
I can make everything protected (I think making them private would make things difficult in the child class). Not sure we'd need getters for all of them since it seems pretty common to make fields only accessible via unparse()
The getParserPos() method is mainly there because it exists in the SqlNode class, and I was trying to conform to that.